packages/flame/lib/src/extensions/vector2.dart

```diff
@@ -50,6 +50,12 @@ extension Vector2Extension on Vector2 {
     }
   }
 
+  /// Scale this vector by the given [scale].
+  void scaleBy(Vector2 scale) {
+    x = x * scale.x;
```

Comment:
We should add a test for this, wdyt?

Reply:
Yes! I can most certainly do that as well